{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using the results from the AAI analysis create graphs of AAI versus O2 consumption for SAG pairs on individual days.\n",
    "\n",
    "### Inputs\n",
    "/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/Summary_files/All_GoM_SAGs_1cell_20kb_decon_531normalized_predresp_rate_GTDBclass.csv\n",
    "/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/aai/GoM_RSG_aai_outputs/aai/aai_summary.tsv\n",
    "\n",
    "### Outputs \n",
    "/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_images/AAI_vs_log10_O2_ratio*.png graph files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "rerun on 2021-12-07 after correcting for salinity of media samples The last bit of this code is not finished"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3.3.3'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from pandas import DataFrame\n",
    "import os\n",
    "import sys\n",
    "import csv\n",
    "from pathlib import Path\n",
    "import seaborn as sns\n",
    "sns.set()\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "plt.show()\n",
    "from collections import defaultdict\n",
    "import os.path as op\n",
    "import glob\n",
    "\n",
    "os.chdir('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/aai/')\n",
    "outdir='/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/aai/'\n",
    "images='/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_images/AAI/'\n",
    "matplotlib.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ",X2,name,Plate,well_type,wga_cp,wgs_run_id,raw_read_count,final_assembly_length,max_contig_length,gc_content,checkM_estimated_completeness,total_CDS_count,hypothetical_cds_fraction,average_cds_length,number_multi_copy,checkM_classification,SSU_classification_1,SSU_classification_2,SSU_classification_3,notes,eventnum,well2,X531.40..488.,X572.27..488.,X692.40..488.,Trigger.Pulse.Width,SSC,FSC,estimated_diameter,PCA_analysis_Action,PCA_analysis_Note,Location,Sample_collection_Date,FACS_date,gate,depth..m.,sample,slope,intercept,normalized_531_value,Predicted_respiration_rate_fmol_O2perhr,lower_resp_bound,upper_resp_bound,SAG,GTDB_classification,GTDBtk_classification\n",
      "0,A01,AH-135-A01,AH-135,1 cell,2.131257507,170717_NB502014_0032_AHMHYCAFXX,428328,378833,26163,29.81076094,22.64,390.0,0.241025641,865.8923077,0,k__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rickettsiales;f__Pelagibacteraceae;g__Candidatus_Pelagibacter;s__Candidatus_Pelagibacter_ubique;unresolved,no SSU regions found,,,,558,A01,5.272114858999999,1.022166126,1.0261963570000001,4229,2.94810846,2.147985029,0.31,,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,AH-135-A01,0.10985,0.18027,0.759412,0.00461047923018511,0.000751783394460428,0.0282747915005819,AH-135-A01,Pelagibacter,d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Pelagibacterales;f__Pelagibacteraceae;g__Pelagibacter;s__Pelagibacter ubique\n",
      "1,A02,AH-135-A02,AH-135,1 cell,1.460275237,AH-135,2502502,1800737,167471,36.02519413,86.23,1622.0,0.332305795,1016.4827369999999,0,k__Bacteria;p__Bacteroidetes;c__Flavobacteriia;o__Flavobacteriales;f__Flavobacteriaceae;g__Olleya,k__Bacteria;p__Bacteroidetes;c__Flavobacteriia;o__Flavobacteriales;f__Flavobacteriaceae;g__Formosa;s__?,,,,4776,A02,128.11647490000001,17.66323766,7.602937936,5631,5.582467095,3.23646415,0.32,keep?,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,AH-135-A02,0.10985,0.18027,14.2539,0.0633555202259773,0.0104689013628961,0.383413865883805,AH-135-A02,Hel1-33-131,d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Flavobacteriales;f__Flavobacteriaceae;g__Hel1-33-131;s__\n",
      "2,A03,AH-135-A03,AH-135,1 cell,1.607216558,AH-135,3997827,856845,84808,32.01185745,74.13,919.0,0.228509249,863.1338411,0,k__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rickettsiales;f__Pelagibacteraceae;g__Candidatus_Pelagibacter;s__,no SSUs were classified,,,,8484,A03,2.3790513090000003,1.021448109,1.022453474,3579,8.547197326000001,1.9996977390000001,0.31,,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,AH-135-A03,0.10985,0.18027,0.44160900000000003,0.0043345558920111,0.0007061226297553331,0.0266078071842538,AH-135-A03,IMCC9063,d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Pelagibacterales;f__Pelagibacteraceae;g__IMCC9063;s__\n",
      "3,A04,AH-135-A04,AH-135,1 cell,1.497029617,AH-135,3618815,1481697,223427,36.51590035,83.69,1467.0,0.194274029,925.8220858999999,0,k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Thiotrichales;p__;c__;o__;f__;g__;s__,k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Oceanospirillales;f__ZD0405;g__?;s__?,,,,11875,A04,72.44115258,6.326266177999999,4.136549099,6522,1.027206396,4.689014146,0.34,,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,AH-135-A04,0.10985,0.18027,8.13793,0.0193198102965454,0.00319412307089112,0.116856821609685,AH-135-A04,Thioglobus,d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__PS1;f__Thioglobaceae;g__Thioglobus;s__\n",
      "4,A05,AH-135-A05,AH-135,1 cell,1.905231842,170717_NB502014_0032_AHMHYCAFXX,664796,382026,35013,34.99081214,19.18,400.0,0.275,840.0125,0,k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__;f__;g__;s__,no SSU regions found,,,,15271,A05,27.29198703,3.79617044,1.023028412,5054,6.550646741,2.425641756,0.31,,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,AH-135-A05,0.10985,0.18027,3.17829,0.00737460383261583,0.00121005224389861,0.0449441600247046,AH-135-A05,SCGC-AAA076-P13,d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__SAR86;f__D2472;g__SCGC-AAA076-P13;s__\n",
      "5,A06,AH-135-A06,AH-135,1 cell,1.695323282,AH-135,5576767,1127973,100196,37.628471600000005,48.49,1113.0,0.253369272,919.7654987,0,k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__;f__;g__;s__,k__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Oceanospirillales;f__SAR86_clade;g__?;s__?,,,,19684,A06,29.52254604,3.298918467,5.183219602,6779,9.458670094,5.656704295,0.35,,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,,0.10985,0.18027,3.42332,0.00773397831791087,0.00126970589334656,0.0471088784696926,,Unclassified,Unclassified\n",
      "6,A07,AH-135-A07,AH-135,1 cell,1.726013961,AH-135,6300693,1141311,307768,29.59999509,92.92,1176.0,0.193027211,919.4285714,0,k__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rickettsiales;f__Pelagibacteraceae;g__Candidatus_Pelagibacter;s__Candidatus_Pelagibacter_ubique;unresolved,k__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__SAR11_clade;f__Surface_1;g__?;s__?,,,,24019,A07,25.52956748,2.3027333580000002,3.7717082000000004,4970,4.007229707,2.0129500940000002,0.31,,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,,0.10985,0.18027,2.98469,0.0071025077288949705,0.0011648938514840401,0.0433049036826375,,Unclassified,Unclassified\n",
      "7,A08,AH-135-A08,AH-135,1 cell,1.77072524,AH-135,4971955,1157221,79268,44.80838146,41.14,1027.0,0.403115871,1008.180136,0,k__Bacteria;p__Bacteroidetes;c__Flavobacteriia;o__Flavobacteriales;f__Cryomorphaceae;g__Owenweeksia;s__Owenweeksia_hongkongensis,no SSUs were classified,,,,27587,A08,84.67093878,8.068621002999999,1.62600024,7130,7.763824034,7.589060032000001,0.38,,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,AH-135-A08,0.10985,0.18027,9.48137,0.0250784074026555,0.00414957711919313,0.15156400273767198,AH-135-A08,UBA10364,d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Flavobacteriales;f__Schleiferiaceae;g__UBA10364;s__UBA10364 sp003045825\n",
      "8,A09,AH-135-A09,AH-135,1 cell,1.689432994,AH-135,4924755,490210,44433,52.73433834,17.24,485.0,0.30103092800000003,879.4164947999999,0,k__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__;s__,k__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhodobacterales;f__Rhodobacteraceae;g__Planktotalea;s__?,,,,31229,A09,529.8111498,55.36369885,39.23086913,8404,18.22557137,23.89438728,0.58,,,EBB Bigelow dock,170412,170412,Respiring cells (RSG-positive),1meter,,0.10985,0.18027,58.38,333.500235884223,40.7539244525828,2729.12139944313,,Unclassified,Unclassified\n"
     ]
    }
   ],
   "source": [
    "!head /mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/Summary_files/All_GoM_SAGs_1cell_20kb_decon_531normalized_predresp_rate_GTDBclass.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>checkM_estimated_completeness</th>\n",
       "      <th>estimated_diameter</th>\n",
       "      <th>Sample_collection_Date</th>\n",
       "      <th>FACS_date</th>\n",
       "      <th>gate</th>\n",
       "      <th>Predicted_respiration_rate_fmol_O2perhr</th>\n",
       "      <th>GTDB_classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>287</th>\n",
       "      <td>AH-141-A07</td>\n",
       "      <td>52.83</td>\n",
       "      <td>0.29</td>\n",
       "      <td>170412</td>\n",
       "      <td>170412</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pelagibacter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>AH-141-A21</td>\n",
       "      <td>79.42</td>\n",
       "      <td>0.30</td>\n",
       "      <td>170412</td>\n",
       "      <td>170412</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pelagibacter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>297</th>\n",
       "      <td>AH-141-A22</td>\n",
       "      <td>65.66</td>\n",
       "      <td>0.35</td>\n",
       "      <td>170412</td>\n",
       "      <td>170412</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pelagibacter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>AH-141-B06</td>\n",
       "      <td>53.63</td>\n",
       "      <td>0.43</td>\n",
       "      <td>170412</td>\n",
       "      <td>170412</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hel1-33-131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>AH-141-B20</td>\n",
       "      <td>62.03</td>\n",
       "      <td>0.37</td>\n",
       "      <td>170412</td>\n",
       "      <td>170412</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Porticoccus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5828</th>\n",
       "      <td>AH-273-O22</td>\n",
       "      <td>58.79</td>\n",
       "      <td>0.51</td>\n",
       "      <td>171102</td>\n",
       "      <td>171102</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA-2691785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5843</th>\n",
       "      <td>AH-273-P16</td>\n",
       "      <td>67.60</td>\n",
       "      <td>0.33</td>\n",
       "      <td>171102</td>\n",
       "      <td>171102</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UBA10364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5845</th>\n",
       "      <td>AH-273-P18</td>\n",
       "      <td>67.28</td>\n",
       "      <td>0.41</td>\n",
       "      <td>171102</td>\n",
       "      <td>171102</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAG-121220-bin8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5846</th>\n",
       "      <td>AH-273-P19</td>\n",
       "      <td>55.29</td>\n",
       "      <td>0.34</td>\n",
       "      <td>171102</td>\n",
       "      <td>171102</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UBA3537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5847</th>\n",
       "      <td>AH-273-P20</td>\n",
       "      <td>80.98</td>\n",
       "      <td>0.23</td>\n",
       "      <td>171102</td>\n",
       "      <td>171102</td>\n",
       "      <td>All prokaryotes (SYTO-9-positive)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pelagibacter_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>526 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            name  checkM_estimated_completeness  estimated_diameter  \\\n",
       "287   AH-141-A07                          52.83                0.29   \n",
       "296   AH-141-A21                          79.42                0.30   \n",
       "297   AH-141-A22                          65.66                0.35   \n",
       "302   AH-141-B06                          53.63                0.43   \n",
       "310   AH-141-B20                          62.03                0.37   \n",
       "...          ...                            ...                 ...   \n",
       "5828  AH-273-O22                          58.79                0.51   \n",
       "5843  AH-273-P16                          67.60                0.33   \n",
       "5845  AH-273-P18                          67.28                0.41   \n",
       "5846  AH-273-P19                          55.29                0.34   \n",
       "5847  AH-273-P20                          80.98                0.23   \n",
       "\n",
       "      Sample_collection_Date  FACS_date                               gate  \\\n",
       "287                   170412     170412  All prokaryotes (SYTO-9-positive)   \n",
       "296                   170412     170412  All prokaryotes (SYTO-9-positive)   \n",
       "297                   170412     170412  All prokaryotes (SYTO-9-positive)   \n",
       "302                   170412     170412  All prokaryotes (SYTO-9-positive)   \n",
       "310                   170412     170412  All prokaryotes (SYTO-9-positive)   \n",
       "...                      ...        ...                                ...   \n",
       "5828                  171102     171102  All prokaryotes (SYTO-9-positive)   \n",
       "5843                  171102     171102  All prokaryotes (SYTO-9-positive)   \n",
       "5845                  171102     171102  All prokaryotes (SYTO-9-positive)   \n",
       "5846                  171102     171102  All prokaryotes (SYTO-9-positive)   \n",
       "5847                  171102     171102  All prokaryotes (SYTO-9-positive)   \n",
       "\n",
       "      Predicted_respiration_rate_fmol_O2perhr GTDB_classification  \n",
       "287                                       NaN        Pelagibacter  \n",
       "296                                       NaN        Pelagibacter  \n",
       "297                                       NaN        Pelagibacter  \n",
       "302                                       NaN         Hel1-33-131  \n",
       "310                                       NaN         Porticoccus  \n",
       "...                                       ...                 ...  \n",
       "5828                                      NaN         GCA-2691785  \n",
       "5843                                      NaN            UBA10364  \n",
       "5845                                      NaN     MAG-121220-bin8  \n",
       "5846                                      NaN             UBA3537  \n",
       "5847                                      NaN      Pelagibacter_A  \n",
       "\n",
       "[526 rows x 8 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns=['name', 'GTDB_classification', 'estimated_diameter', 'Predicted_respiration_rate_fmol_O2perhr',\n",
    "        'Sample_collection_Date', 'gate', 'FACS_date', 'checkM_estimated_completeness']\n",
    "SAG_key=pd.read_csv('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/Summary_files/All_GoM_SAGs_1cell_20kb_decon_531normalized_predresp_rate_GTDBclass.csv',\n",
    "                   usecols=columns)\n",
    "SAG_key=SAG_key[SAG_key['gate']=='All prokaryotes (SYTO-9-positive)']\n",
    "SAG_key=SAG_key[SAG_key['Sample_collection_Date']==SAG_key['FACS_date']]\n",
    "SAG_key=SAG_key[SAG_key['checkM_estimated_completeness']>=50]\n",
    "SAG_list=SAG_key['name'].tolist()\n",
    "SAG_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#read result file from RSG_ANI analysis and parse it to remove AAI=0 and add SAG identifiers\n",
    "\n",
    "aai=pd.read_csv('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/aai/GoM_Syto9_outputs/aai/aai_summary.tsv',\n",
    "                sep='\\t')\n",
    "aai=aai[aai['Mean AAI'] != 0.0]\n",
    "aai['SAG_A']=aai['Genome A'].str[:10]\n",
    "aai['SAG_B']=aai['Genome B'].str[:10]\n",
    "\n",
    "aai=aai[aai['SAG_A'].isin(SAG_list)]\n",
    "aai=aai[aai['SAG_B'].isin(SAG_list)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome A</th>\n",
       "      <th>Genes in A</th>\n",
       "      <th>Genome B</th>\n",
       "      <th>Genes in B</th>\n",
       "      <th># orthologous genes</th>\n",
       "      <th>Mean AAI</th>\n",
       "      <th>Std AAI</th>\n",
       "      <th>Orthologous fraction (OF)</th>\n",
       "      <th>SAG_A</th>\n",
       "      <th>SAG_B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8605</th>\n",
       "      <td>AH-660-L02_contigs</td>\n",
       "      <td>1126</td>\n",
       "      <td>AH-229-N19_contigs</td>\n",
       "      <td>1154</td>\n",
       "      <td>139</td>\n",
       "      <td>41.17</td>\n",
       "      <td>8.20</td>\n",
       "      <td>12.34</td>\n",
       "      <td>AH-660-L02</td>\n",
       "      <td>AH-229-N19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8622</th>\n",
       "      <td>AH-660-L02_contigs</td>\n",
       "      <td>1126</td>\n",
       "      <td>AH-229-K22_contigs</td>\n",
       "      <td>1205</td>\n",
       "      <td>131</td>\n",
       "      <td>42.55</td>\n",
       "      <td>7.74</td>\n",
       "      <td>11.63</td>\n",
       "      <td>AH-660-L02</td>\n",
       "      <td>AH-229-K22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8630</th>\n",
       "      <td>AH-660-L02_contigs</td>\n",
       "      <td>1126</td>\n",
       "      <td>AH-273-P20_contigs</td>\n",
       "      <td>1052</td>\n",
       "      <td>123</td>\n",
       "      <td>43.59</td>\n",
       "      <td>9.71</td>\n",
       "      <td>11.69</td>\n",
       "      <td>AH-660-L02</td>\n",
       "      <td>AH-273-P20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8632</th>\n",
       "      <td>AH-660-L02_contigs</td>\n",
       "      <td>1126</td>\n",
       "      <td>AH-541-O07_contigs</td>\n",
       "      <td>937</td>\n",
       "      <td>104</td>\n",
       "      <td>44.71</td>\n",
       "      <td>9.68</td>\n",
       "      <td>11.10</td>\n",
       "      <td>AH-660-L02</td>\n",
       "      <td>AH-541-O07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8633</th>\n",
       "      <td>AH-660-L02_contigs</td>\n",
       "      <td>1126</td>\n",
       "      <td>AH-660-L16_contigs</td>\n",
       "      <td>942</td>\n",
       "      <td>99</td>\n",
       "      <td>43.94</td>\n",
       "      <td>9.43</td>\n",
       "      <td>10.51</td>\n",
       "      <td>AH-660-L02</td>\n",
       "      <td>AH-660-L16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519509</th>\n",
       "      <td>AH-660-J19_contigs</td>\n",
       "      <td>1145</td>\n",
       "      <td>AH-141-C19_contigs</td>\n",
       "      <td>1477</td>\n",
       "      <td>132</td>\n",
       "      <td>42.75</td>\n",
       "      <td>8.12</td>\n",
       "      <td>11.53</td>\n",
       "      <td>AH-660-J19</td>\n",
       "      <td>AH-141-C19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519512</th>\n",
       "      <td>AH-660-J19_contigs</td>\n",
       "      <td>1145</td>\n",
       "      <td>AH-229-C19_contigs</td>\n",
       "      <td>885</td>\n",
       "      <td>578</td>\n",
       "      <td>78.13</td>\n",
       "      <td>12.84</td>\n",
       "      <td>65.31</td>\n",
       "      <td>AH-660-J19</td>\n",
       "      <td>AH-229-C19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519516</th>\n",
       "      <td>AH-660-J19_contigs</td>\n",
       "      <td>1145</td>\n",
       "      <td>AH-541-G04_contigs</td>\n",
       "      <td>1574</td>\n",
       "      <td>300</td>\n",
       "      <td>47.49</td>\n",
       "      <td>11.39</td>\n",
       "      <td>26.20</td>\n",
       "      <td>AH-660-J19</td>\n",
       "      <td>AH-541-G04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519521</th>\n",
       "      <td>AH-660-J19_contigs</td>\n",
       "      <td>1145</td>\n",
       "      <td>AH-273-E10_contigs</td>\n",
       "      <td>1399</td>\n",
       "      <td>369</td>\n",
       "      <td>46.69</td>\n",
       "      <td>10.93</td>\n",
       "      <td>32.23</td>\n",
       "      <td>AH-660-J19</td>\n",
       "      <td>AH-273-E10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1519523</th>\n",
       "      <td>AH-660-J19_contigs</td>\n",
       "      <td>1145</td>\n",
       "      <td>AH-709-P23_contigs</td>\n",
       "      <td>1631</td>\n",
       "      <td>179</td>\n",
       "      <td>43.16</td>\n",
       "      <td>9.06</td>\n",
       "      <td>15.63</td>\n",
       "      <td>AH-660-J19</td>\n",
       "      <td>AH-709-P23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138075 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Genome A  Genes in A            Genome B  Genes in B  \\\n",
       "8605     AH-660-L02_contigs        1126  AH-229-N19_contigs        1154   \n",
       "8622     AH-660-L02_contigs        1126  AH-229-K22_contigs        1205   \n",
       "8630     AH-660-L02_contigs        1126  AH-273-P20_contigs        1052   \n",
       "8632     AH-660-L02_contigs        1126  AH-541-O07_contigs         937   \n",
       "8633     AH-660-L02_contigs        1126  AH-660-L16_contigs         942   \n",
       "...                     ...         ...                 ...         ...   \n",
       "1519509  AH-660-J19_contigs        1145  AH-141-C19_contigs        1477   \n",
       "1519512  AH-660-J19_contigs        1145  AH-229-C19_contigs         885   \n",
       "1519516  AH-660-J19_contigs        1145  AH-541-G04_contigs        1574   \n",
       "1519521  AH-660-J19_contigs        1145  AH-273-E10_contigs        1399   \n",
       "1519523  AH-660-J19_contigs        1145  AH-709-P23_contigs        1631   \n",
       "\n",
       "         # orthologous genes  Mean AAI  Std AAI  Orthologous fraction (OF)  \\\n",
       "8605                     139     41.17     8.20                      12.34   \n",
       "8622                     131     42.55     7.74                      11.63   \n",
       "8630                     123     43.59     9.71                      11.69   \n",
       "8632                     104     44.71     9.68                      11.10   \n",
       "8633                      99     43.94     9.43                      10.51   \n",
       "...                      ...       ...      ...                        ...   \n",
       "1519509                  132     42.75     8.12                      11.53   \n",
       "1519512                  578     78.13    12.84                      65.31   \n",
       "1519516                  300     47.49    11.39                      26.20   \n",
       "1519521                  369     46.69    10.93                      32.23   \n",
       "1519523                  179     43.16     9.06                      15.63   \n",
       "\n",
       "              SAG_A       SAG_B  \n",
       "8605     AH-660-L02  AH-229-N19  \n",
       "8622     AH-660-L02  AH-229-K22  \n",
       "8630     AH-660-L02  AH-273-P20  \n",
       "8632     AH-660-L02  AH-541-O07  \n",
       "8633     AH-660-L02  AH-660-L16  \n",
       "...             ...         ...  \n",
       "1519509  AH-660-J19  AH-141-C19  \n",
       "1519512  AH-660-J19  AH-229-C19  \n",
       "1519516  AH-660-J19  AH-541-G04  \n",
       "1519521  AH-660-J19  AH-273-E10  \n",
       "1519523  AH-660-J19  AH-709-P23  \n",
       "\n",
       "[138075 rows x 10 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aai"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-058bbc58fb7b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mSyto9hist\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhistplot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'Mean AAI'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0max\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0max\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbinwidth\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mAAIhist\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mAAIhist\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_title\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Sampling Date \"\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mfig\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mAAIhist\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_figure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mfig\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msavefig\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mimages\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'SYTO9_AAI_histogram_'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m\".pdf\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df' is not defined"
     ]
    }
   ],
   "source": [
    "# This bit of code was not finished.\n",
    "\n",
    "Syto9hist=sns.histplot(data=aai, x='Mean AAI', ax=ax, binwidth=1)\n",
    "AAIhist=AAIhist.set_title(\"Sampling Date \"+i)\n",
    "fig=AAIhist.get_figure()\n",
    "fig.savefig(images+'SYTO9_AAI_histogram_'+\".pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
