{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "URL: https://github.com/christophertbrown/iRep\n",
    "\n",
    "### Instructions\n",
    "#### Inputs\n",
    "iRep requires high-quality draft genomes (>=75% complete, <=175 fragments/Mbp sequence, and <=2% contamination). iRep will only make use of scaffolds â‰¥5 Kbp in length, thus it is important to make sure that estimates of genome completion are calculated based on scaffolds that pass this threshold\n",
    "\n",
    "I also want to select just the most complete SAG per genus\n",
    "\n",
    "The second set of inputs are mapping files in SAM format with the --reorder flag, which should be generated by mapping DNA sequencing reads against assembled genomes using Bowtie2. In this case, provide the path to each SAM file generated for each sample"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "rerun on 2021-12-07 after correcting for salinity of media samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3.3.3'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from pandas import DataFrame\n",
    "import os\n",
    "import sys\n",
    "import csv\n",
    "from pathlib import Path\n",
    "import seaborn as sns\n",
    "sns.set()\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "plt.show()\n",
    "from collections import defaultdict\n",
    "import os.path as op\n",
    "import glob\n",
    "\n",
    "os.chdir('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP')\n",
    "outdir=('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP')\n",
    "matplotlib.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>X2</th>\n",
       "      <th>name</th>\n",
       "      <th>Plate</th>\n",
       "      <th>well_type</th>\n",
       "      <th>wga_cp</th>\n",
       "      <th>wgs_run_id</th>\n",
       "      <th>raw_read_count</th>\n",
       "      <th>final_assembly_length</th>\n",
       "      <th>max_contig_length</th>\n",
       "      <th>gc_content</th>\n",
       "      <th>...</th>\n",
       "      <th>sample</th>\n",
       "      <th>slope</th>\n",
       "      <th>intercept</th>\n",
       "      <th>normalized_531_value</th>\n",
       "      <th>Predicted_respiration_rate_fmol_O2perhr</th>\n",
       "      <th>lower_resp_bound</th>\n",
       "      <th>upper_resp_bound</th>\n",
       "      <th>SAG</th>\n",
       "      <th>GTDB_classification</th>\n",
       "      <th>GTDBtk_classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A02</td>\n",
       "      <td>AH-135-A02</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.460275237</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>2502502</td>\n",
       "      <td>1800737</td>\n",
       "      <td>167471</td>\n",
       "      <td>36.025194</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-135-A02</td>\n",
       "      <td>0.10985</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>14.25390</td>\n",
       "      <td>0.063356</td>\n",
       "      <td>0.010469</td>\n",
       "      <td>0.383414</td>\n",
       "      <td>AH-135-A02</td>\n",
       "      <td>Hel1-33-131</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A04</td>\n",
       "      <td>AH-135-A04</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.497029617</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>3618815</td>\n",
       "      <td>1481697</td>\n",
       "      <td>223427</td>\n",
       "      <td>36.515900</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-135-A04</td>\n",
       "      <td>0.10985</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>8.13793</td>\n",
       "      <td>0.019320</td>\n",
       "      <td>0.003194</td>\n",
       "      <td>0.116857</td>\n",
       "      <td>AH-135-A04</td>\n",
       "      <td>Thioglobus</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A07</td>\n",
       "      <td>AH-135-A07</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.726013961</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>6300693</td>\n",
       "      <td>1141311</td>\n",
       "      <td>307768</td>\n",
       "      <td>29.599995</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.10985</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>2.98469</td>\n",
       "      <td>0.007103</td>\n",
       "      <td>0.001165</td>\n",
       "      <td>0.043305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unclassified</td>\n",
       "      <td>Unclassified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>A14</td>\n",
       "      <td>AH-135-A14</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.673261954</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>5754909</td>\n",
       "      <td>1679721</td>\n",
       "      <td>178475</td>\n",
       "      <td>34.100365</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-135-A14</td>\n",
       "      <td>0.10985</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>15.29750</td>\n",
       "      <td>0.077588</td>\n",
       "      <td>0.012806</td>\n",
       "      <td>0.470101</td>\n",
       "      <td>AH-135-A14</td>\n",
       "      <td>Polaribacter</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>A18</td>\n",
       "      <td>AH-135-A18</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.618199506</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>5882001</td>\n",
       "      <td>2105486</td>\n",
       "      <td>314702</td>\n",
       "      <td>35.388552</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-135-A18</td>\n",
       "      <td>0.10985</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>16.33200</td>\n",
       "      <td>0.094850</td>\n",
       "      <td>0.015631</td>\n",
       "      <td>0.575543</td>\n",
       "      <td>AH-135-A18</td>\n",
       "      <td>MS024-2A</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5784</th>\n",
       "      <td>M13</td>\n",
       "      <td>AH-273-M13</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.517333187</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3412078</td>\n",
       "      <td>1590989</td>\n",
       "      <td>127065</td>\n",
       "      <td>33.726883</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-273-M13</td>\n",
       "      <td>0.10055</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>34.71660</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-M13</td>\n",
       "      <td>Hel1-33-131</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5794</th>\n",
       "      <td>N03</td>\n",
       "      <td>AH-273-N03</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.587519407</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3210354</td>\n",
       "      <td>1044397</td>\n",
       "      <td>166130</td>\n",
       "      <td>29.393899</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-273-N03</td>\n",
       "      <td>0.10055</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>11.93090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-N03</td>\n",
       "      <td>Pelagibacter</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5803</th>\n",
       "      <td>N16</td>\n",
       "      <td>AH-273-N16</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.34815115</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3136144</td>\n",
       "      <td>1097658</td>\n",
       "      <td>333887</td>\n",
       "      <td>29.903030</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-273-N16</td>\n",
       "      <td>0.10055</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>11.69670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-N16</td>\n",
       "      <td>Pelagibacter</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5807</th>\n",
       "      <td>N21</td>\n",
       "      <td>AH-273-N21</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.226552412</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3108541</td>\n",
       "      <td>2073803</td>\n",
       "      <td>192110</td>\n",
       "      <td>44.518790</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-273-N21</td>\n",
       "      <td>0.10055</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>29.33790</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-N21</td>\n",
       "      <td>MB11C04</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5847</th>\n",
       "      <td>P20</td>\n",
       "      <td>AH-273-P20</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.347279162</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3110851</td>\n",
       "      <td>1000803</td>\n",
       "      <td>226113</td>\n",
       "      <td>29.565959</td>\n",
       "      <td>...</td>\n",
       "      <td>AH-273-P20</td>\n",
       "      <td>0.10055</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>6.11717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-P20</td>\n",
       "      <td>Pelagibacter_A</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>373 rows Ã— 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       X2        name   Plate well_type       wga_cp wgs_run_id  \\\n",
       "1     A02  AH-135-A02  AH-135    1 cell  1.460275237     AH-135   \n",
       "3     A04  AH-135-A04  AH-135    1 cell  1.497029617     AH-135   \n",
       "6     A07  AH-135-A07  AH-135    1 cell  1.726013961     AH-135   \n",
       "12    A14  AH-135-A14  AH-135    1 cell  1.673261954     AH-135   \n",
       "16    A18  AH-135-A18  AH-135    1 cell  1.618199506     AH-135   \n",
       "...   ...         ...     ...       ...          ...        ...   \n",
       "5784  M13  AH-273-M13  AH-273    1 cell  1.517333187     AH-273   \n",
       "5794  N03  AH-273-N03  AH-273    1 cell  1.587519407     AH-273   \n",
       "5803  N16  AH-273-N16  AH-273    1 cell   1.34815115     AH-273   \n",
       "5807  N21  AH-273-N21  AH-273    1 cell  1.226552412     AH-273   \n",
       "5847  P20  AH-273-P20  AH-273    1 cell  1.347279162     AH-273   \n",
       "\n",
       "      raw_read_count  final_assembly_length  max_contig_length  gc_content  \\\n",
       "1            2502502                1800737             167471   36.025194   \n",
       "3            3618815                1481697             223427   36.515900   \n",
       "6            6300693                1141311             307768   29.599995   \n",
       "12           5754909                1679721             178475   34.100365   \n",
       "16           5882001                2105486             314702   35.388552   \n",
       "...              ...                    ...                ...         ...   \n",
       "5784         3412078                1590989             127065   33.726883   \n",
       "5794         3210354                1044397             166130   29.393899   \n",
       "5803         3136144                1097658             333887   29.903030   \n",
       "5807         3108541                2073803             192110   44.518790   \n",
       "5847         3110851                1000803             226113   29.565959   \n",
       "\n",
       "      ...      sample    slope  intercept  normalized_531_value  \\\n",
       "1     ...  AH-135-A02  0.10985    0.18027              14.25390   \n",
       "3     ...  AH-135-A04  0.10985    0.18027               8.13793   \n",
       "6     ...         NaN  0.10985    0.18027               2.98469   \n",
       "12    ...  AH-135-A14  0.10985    0.18027              15.29750   \n",
       "16    ...  AH-135-A18  0.10985    0.18027              16.33200   \n",
       "...   ...         ...      ...        ...                   ...   \n",
       "5784  ...  AH-273-M13  0.10055    0.32122              34.71660   \n",
       "5794  ...  AH-273-N03  0.10055    0.32122              11.93090   \n",
       "5803  ...  AH-273-N16  0.10055    0.32122              11.69670   \n",
       "5807  ...  AH-273-N21  0.10055    0.32122              29.33790   \n",
       "5847  ...  AH-273-P20  0.10055    0.32122               6.11717   \n",
       "\n",
       "     Predicted_respiration_rate_fmol_O2perhr lower_resp_bound  \\\n",
       "1                                   0.063356         0.010469   \n",
       "3                                   0.019320         0.003194   \n",
       "6                                   0.007103         0.001165   \n",
       "12                                  0.077588         0.012806   \n",
       "16                                  0.094850         0.015631   \n",
       "...                                      ...              ...   \n",
       "5784                                     NaN              NaN   \n",
       "5794                                     NaN              NaN   \n",
       "5803                                     NaN              NaN   \n",
       "5807                                     NaN              NaN   \n",
       "5847                                     NaN              NaN   \n",
       "\n",
       "     upper_resp_bound         SAG  GTDB_classification  \\\n",
       "1            0.383414  AH-135-A02          Hel1-33-131   \n",
       "3            0.116857  AH-135-A04           Thioglobus   \n",
       "6            0.043305         NaN         Unclassified   \n",
       "12           0.470101  AH-135-A14         Polaribacter   \n",
       "16           0.575543  AH-135-A18             MS024-2A   \n",
       "...               ...         ...                  ...   \n",
       "5784              NaN  AH-273-M13          Hel1-33-131   \n",
       "5794              NaN  AH-273-N03         Pelagibacter   \n",
       "5803              NaN  AH-273-N16         Pelagibacter   \n",
       "5807              NaN  AH-273-N21              MB11C04   \n",
       "5847              NaN  AH-273-P20       Pelagibacter_A   \n",
       "\n",
       "                                  GTDBtk_classification  \n",
       "1     d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...  \n",
       "3     d__Bacteria;p__Proteobacteria;c__Gammaproteoba...  \n",
       "6                                          Unclassified  \n",
       "12    d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...  \n",
       "16    d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...  \n",
       "...                                                 ...  \n",
       "5784  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...  \n",
       "5794  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  \n",
       "5803  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  \n",
       "5807  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...  \n",
       "5847  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...  \n",
       "\n",
       "[373 rows x 46 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAG_data=pd.read_csv('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/Summary_files/All_GoM_SAGs_1cell_20kb_decon_531normalized_predresp_rate_GTDBclass.csv',\n",
    "                    index_col=0)\n",
    "SAG_data=SAG_data[SAG_data['checkM_estimated_completeness'] >= 75]\n",
    "\n",
    "SAGs75perc=SAG_data['name'].to_list()\n",
    "\n",
    "MyFile=open('75perc_SAGs.txt','w')\n",
    "\n",
    "for element in SAGs75perc:\n",
    "     MyFile.write(element)\n",
    "     MyFile.write('\\n')\n",
    "MyFile.close()\n",
    "SAG_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#If file does not exist do the things to create it\n",
    "#\n",
    "if os.path.isfile('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP/75perc_complete_SAGs/75perc_stats.tab')== False:        \n",
    "    os.chdir('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP/')\n",
    "    !grep -f 75perc_SAGs.txt /mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/Summary_files/SAG_contigs.txt > 75perc_complete_SAGs/75perc_SAG_contigs.txt\n",
    "\n",
    "    os.chdir('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP/75perc_complete_SAGs')\n",
    "\n",
    "    !while read file; do cp \"$file\" . ; done < 75perc_SAG_contigs.txt\n",
    "    !seqkit stats *.fasta -T > 75perc_stats.tab\n",
    "    os.chdir('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_seqs</th>\n",
       "      <th>SAG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>50</td>\n",
       "      <td>AH-135-A02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AH-135-A04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>59</td>\n",
       "      <td>AH-135-A14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>45</td>\n",
       "      <td>AH-135-A18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>72</td>\n",
       "      <td>AH-135-B03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>44</td>\n",
       "      <td>AH-709-O07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356</th>\n",
       "      <td>47</td>\n",
       "      <td>AH-709-P04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357</th>\n",
       "      <td>39</td>\n",
       "      <td>AH-709-P06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>51</td>\n",
       "      <td>AH-709-P13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>32</td>\n",
       "      <td>AH-709-P17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>360 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     num_seqs         SAG\n",
       "0          50  AH-135-A02\n",
       "1          46  AH-135-A04\n",
       "2          59  AH-135-A14\n",
       "3          45  AH-135-A18\n",
       "4          72  AH-135-B03\n",
       "..        ...         ...\n",
       "355        44  AH-709-O07\n",
       "356        47  AH-709-P04\n",
       "357        39  AH-709-P06\n",
       "358        51  AH-709-P13\n",
       "359        32  AH-709-P17\n",
       "\n",
       "[360 rows x 2 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "stats=pd.read_csv('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP/75perc_complete_SAGs/75perc_stats.tab',\n",
    "                 sep='\\t')\n",
    "stats['SAG']=stats['file'].str[:10]\n",
    "contigs=stats[['num_seqs', 'SAG']].copy()\n",
    "contigs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>X2</th>\n",
       "      <th>name</th>\n",
       "      <th>Plate</th>\n",
       "      <th>well_type</th>\n",
       "      <th>wga_cp</th>\n",
       "      <th>wgs_run_id</th>\n",
       "      <th>raw_read_count</th>\n",
       "      <th>final_assembly_length</th>\n",
       "      <th>max_contig_length</th>\n",
       "      <th>gc_content</th>\n",
       "      <th>...</th>\n",
       "      <th>intercept</th>\n",
       "      <th>normalized_531_value</th>\n",
       "      <th>Predicted_respiration_rate_fmol_O2perhr</th>\n",
       "      <th>lower_resp_bound</th>\n",
       "      <th>upper_resp_bound</th>\n",
       "      <th>SAG</th>\n",
       "      <th>GTDB_classification</th>\n",
       "      <th>GTDBtk_classification</th>\n",
       "      <th>num_seqs</th>\n",
       "      <th>frag_per_Mbp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A02</td>\n",
       "      <td>AH-135-A02</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.460275237</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>2502502</td>\n",
       "      <td>1800737</td>\n",
       "      <td>167471</td>\n",
       "      <td>36.025194</td>\n",
       "      <td>...</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>14.25390</td>\n",
       "      <td>0.063356</td>\n",
       "      <td>0.010469</td>\n",
       "      <td>0.383414</td>\n",
       "      <td>AH-135-A02</td>\n",
       "      <td>Hel1-33-131</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>50.0</td>\n",
       "      <td>27.766409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A04</td>\n",
       "      <td>AH-135-A04</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.497029617</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>3618815</td>\n",
       "      <td>1481697</td>\n",
       "      <td>223427</td>\n",
       "      <td>36.515900</td>\n",
       "      <td>...</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>8.13793</td>\n",
       "      <td>0.019320</td>\n",
       "      <td>0.003194</td>\n",
       "      <td>0.116857</td>\n",
       "      <td>AH-135-A04</td>\n",
       "      <td>Thioglobus</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>46.0</td>\n",
       "      <td>31.045484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A14</td>\n",
       "      <td>AH-135-A14</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.673261954</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>5754909</td>\n",
       "      <td>1679721</td>\n",
       "      <td>178475</td>\n",
       "      <td>34.100365</td>\n",
       "      <td>...</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>15.29750</td>\n",
       "      <td>0.077588</td>\n",
       "      <td>0.012806</td>\n",
       "      <td>0.470101</td>\n",
       "      <td>AH-135-A14</td>\n",
       "      <td>Polaribacter</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>59.0</td>\n",
       "      <td>35.124881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A18</td>\n",
       "      <td>AH-135-A18</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.618199506</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>5882001</td>\n",
       "      <td>2105486</td>\n",
       "      <td>314702</td>\n",
       "      <td>35.388552</td>\n",
       "      <td>...</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>16.33200</td>\n",
       "      <td>0.094850</td>\n",
       "      <td>0.015631</td>\n",
       "      <td>0.575543</td>\n",
       "      <td>AH-135-A18</td>\n",
       "      <td>MS024-2A</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>45.0</td>\n",
       "      <td>21.372738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>B03</td>\n",
       "      <td>AH-135-B03</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.359260806</td>\n",
       "      <td>AH-135</td>\n",
       "      <td>3832623</td>\n",
       "      <td>2566741</td>\n",
       "      <td>174985</td>\n",
       "      <td>47.661763</td>\n",
       "      <td>...</td>\n",
       "      <td>0.18027</td>\n",
       "      <td>11.48640</td>\n",
       "      <td>0.037016</td>\n",
       "      <td>0.006126</td>\n",
       "      <td>0.223657</td>\n",
       "      <td>AH-135-B03</td>\n",
       "      <td>Pseudohongiellaceae</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>72.0</td>\n",
       "      <td>28.051136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>M13</td>\n",
       "      <td>AH-273-M13</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.517333187</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3412078</td>\n",
       "      <td>1590989</td>\n",
       "      <td>127065</td>\n",
       "      <td>33.726883</td>\n",
       "      <td>...</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>34.71660</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-M13</td>\n",
       "      <td>Hel1-33-131</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>74.0</td>\n",
       "      <td>46.511949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>N03</td>\n",
       "      <td>AH-273-N03</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.587519407</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3210354</td>\n",
       "      <td>1044397</td>\n",
       "      <td>166130</td>\n",
       "      <td>29.393899</td>\n",
       "      <td>...</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>11.93090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-N03</td>\n",
       "      <td>Pelagibacter</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>47.0</td>\n",
       "      <td>45.002044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>N16</td>\n",
       "      <td>AH-273-N16</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.34815115</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3136144</td>\n",
       "      <td>1097658</td>\n",
       "      <td>333887</td>\n",
       "      <td>29.903030</td>\n",
       "      <td>...</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>11.69670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-N16</td>\n",
       "      <td>Pelagibacter</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>29.0</td>\n",
       "      <td>26.419887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>N21</td>\n",
       "      <td>AH-273-N21</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.226552412</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3108541</td>\n",
       "      <td>2073803</td>\n",
       "      <td>192110</td>\n",
       "      <td>44.518790</td>\n",
       "      <td>...</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>29.33790</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-N21</td>\n",
       "      <td>MB11C04</td>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>112.0</td>\n",
       "      <td>54.007059</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>372</th>\n",
       "      <td>P20</td>\n",
       "      <td>AH-273-P20</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>1 cell</td>\n",
       "      <td>1.347279162</td>\n",
       "      <td>AH-273</td>\n",
       "      <td>3110851</td>\n",
       "      <td>1000803</td>\n",
       "      <td>226113</td>\n",
       "      <td>29.565959</td>\n",
       "      <td>...</td>\n",
       "      <td>0.32122</td>\n",
       "      <td>6.11717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AH-273-P20</td>\n",
       "      <td>Pelagibacter_A</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>31.974325</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>360 rows Ã— 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      X2        name   Plate well_type       wga_cp wgs_run_id  \\\n",
       "0    A02  AH-135-A02  AH-135    1 cell  1.460275237     AH-135   \n",
       "1    A04  AH-135-A04  AH-135    1 cell  1.497029617     AH-135   \n",
       "3    A14  AH-135-A14  AH-135    1 cell  1.673261954     AH-135   \n",
       "4    A18  AH-135-A18  AH-135    1 cell  1.618199506     AH-135   \n",
       "5    B03  AH-135-B03  AH-135    1 cell  1.359260806     AH-135   \n",
       "..   ...         ...     ...       ...          ...        ...   \n",
       "368  M13  AH-273-M13  AH-273    1 cell  1.517333187     AH-273   \n",
       "369  N03  AH-273-N03  AH-273    1 cell  1.587519407     AH-273   \n",
       "370  N16  AH-273-N16  AH-273    1 cell   1.34815115     AH-273   \n",
       "371  N21  AH-273-N21  AH-273    1 cell  1.226552412     AH-273   \n",
       "372  P20  AH-273-P20  AH-273    1 cell  1.347279162     AH-273   \n",
       "\n",
       "     raw_read_count  final_assembly_length  max_contig_length  gc_content  \\\n",
       "0           2502502                1800737             167471   36.025194   \n",
       "1           3618815                1481697             223427   36.515900   \n",
       "3           5754909                1679721             178475   34.100365   \n",
       "4           5882001                2105486             314702   35.388552   \n",
       "5           3832623                2566741             174985   47.661763   \n",
       "..              ...                    ...                ...         ...   \n",
       "368         3412078                1590989             127065   33.726883   \n",
       "369         3210354                1044397             166130   29.393899   \n",
       "370         3136144                1097658             333887   29.903030   \n",
       "371         3108541                2073803             192110   44.518790   \n",
       "372         3110851                1000803             226113   29.565959   \n",
       "\n",
       "     ...  intercept  normalized_531_value  \\\n",
       "0    ...    0.18027              14.25390   \n",
       "1    ...    0.18027               8.13793   \n",
       "3    ...    0.18027              15.29750   \n",
       "4    ...    0.18027              16.33200   \n",
       "5    ...    0.18027              11.48640   \n",
       "..   ...        ...                   ...   \n",
       "368  ...    0.32122              34.71660   \n",
       "369  ...    0.32122              11.93090   \n",
       "370  ...    0.32122              11.69670   \n",
       "371  ...    0.32122              29.33790   \n",
       "372  ...    0.32122               6.11717   \n",
       "\n",
       "     Predicted_respiration_rate_fmol_O2perhr  lower_resp_bound  \\\n",
       "0                                   0.063356          0.010469   \n",
       "1                                   0.019320          0.003194   \n",
       "3                                   0.077588          0.012806   \n",
       "4                                   0.094850          0.015631   \n",
       "5                                   0.037016          0.006126   \n",
       "..                                       ...               ...   \n",
       "368                                      NaN               NaN   \n",
       "369                                      NaN               NaN   \n",
       "370                                      NaN               NaN   \n",
       "371                                      NaN               NaN   \n",
       "372                                      NaN               NaN   \n",
       "\n",
       "    upper_resp_bound         SAG  GTDB_classification  \\\n",
       "0           0.383414  AH-135-A02          Hel1-33-131   \n",
       "1           0.116857  AH-135-A04           Thioglobus   \n",
       "3           0.470101  AH-135-A14         Polaribacter   \n",
       "4           0.575543  AH-135-A18             MS024-2A   \n",
       "5           0.223657  AH-135-B03  Pseudohongiellaceae   \n",
       "..               ...         ...                  ...   \n",
       "368              NaN  AH-273-M13          Hel1-33-131   \n",
       "369              NaN  AH-273-N03         Pelagibacter   \n",
       "370              NaN  AH-273-N16         Pelagibacter   \n",
       "371              NaN  AH-273-N21              MB11C04   \n",
       "372              NaN  AH-273-P20       Pelagibacter_A   \n",
       "\n",
       "                                 GTDBtk_classification  num_seqs frag_per_Mbp  \n",
       "0    d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...      50.0    27.766409  \n",
       "1    d__Bacteria;p__Proteobacteria;c__Gammaproteoba...      46.0    31.045484  \n",
       "3    d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...      59.0    35.124881  \n",
       "4    d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...      45.0    21.372738  \n",
       "5    d__Bacteria;p__Proteobacteria;c__Gammaproteoba...      72.0    28.051136  \n",
       "..                                                 ...       ...          ...  \n",
       "368  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...      74.0    46.511949  \n",
       "369  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...      47.0    45.002044  \n",
       "370  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...      29.0    26.419887  \n",
       "371  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...     112.0    54.007059  \n",
       "372  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...      32.0    31.974325  \n",
       "\n",
       "[360 rows x 48 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAG_data=pd.read_csv('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/Summary_files/All_GoM_SAGs_1cell_20kb_decon_531normalized_predresp_rate_GTDBclass.csv',\n",
    "                    index_col=0)\n",
    "SAG_data=SAG_data[SAG_data['checkM_estimated_completeness'] >= 75]\n",
    "SAG_data=SAG_data.merge(contigs, how='left', on='SAG', )\n",
    "\n",
    "SAG_data['frag_per_Mbp']=SAG_data['num_seqs']/(SAG_data['final_assembly_length']/1000000)\n",
    "\n",
    "SAG_data=SAG_data[SAG_data['frag_per_Mbp'] <= 175]\n",
    "SAG_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# I now need to figure out how to get a list of the most complete genome per genus. only one per value\n",
    "SAG_data.sort_values(by=['GTDB_classification','checkM_estimated_completeness'], ascending = False).drop_duplicates(subset = ['GTDB_classification'], keep='first')\n",
    "\n",
    "sag_list = list(SAG_data.sort_values(by=['GTDB_classification','checkM_estimated_completeness'], ascending = False).drop_duplicates(subset = ['GTDB_classification'], keep='first')['name'])\n",
    "\n",
    "genus_SAGs=SAG_data[SAG_data['name'].isin(sag_list)]\n",
    "\n",
    "genus_SAGs\n",
    "\n",
    "MyFile=open('largest_SAGs_per_genus.txt','w')\n",
    "\n",
    "for element in sag_list:\n",
    "     MyFile.write(element)\n",
    "     MyFile.write('\\n')\n",
    "MyFile.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#copy the files for iREP into a new folder\n",
    "\n",
    "#!mkdir iREP_SAGs\n",
    "!grep -f largest_SAGs_per_genus.txt /mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/Summary_files/SAG_contigs.txt > largest_SAGs_per_genus_path.txt\n",
    "\n",
    "os.chdir('/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP/iREP_SAGs/')\n",
    "!while read file; do cp \"$file\" . ; done < ../largest_SAGs_per_genus_path.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "77 /mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP/largest_SAGs_per_genus_path.txt\n"
     ]
    }
   ],
   "source": [
    "!wc -l /mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/20210325_GoM_recluster_analysis/iREP/largest_SAGs_per_genus_path.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I think that at this point I am now at the point where I need to recruit the metagenomes to the genomes for the 78 genera that I have a SAG that fulfills the basic iREP requirements.\n",
    "\n",
    "### Questions\n",
    "1. Should I expand this to the largest genome for every genus, recruit to all SAGs? (I think that this would reduce the recruitment of the rest of the SAGs). Recruit to only the ~78 SAGs that I just identified This might result in overrecruitment of reads to these genomes. Does this matter for this analysis?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n",
      "\u001b[33m[WARN]\u001b[0m you may switch on flag -g/--remove-gaps to remove spaces\n"
     ]
    }
   ],
   "source": [
    "#!mkdir iRep_SAGs_5kb\n",
    "!for file in iREP_SAGs/*.fasta; do seqkit seq -m 5000 $file -o ${file/%.fasta/_5kb.fasta}; done\n",
    "!cp iREP_SAGs/*5kb.fasta iRep_SAGs_5kb/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "transform sequences (revserse, complement, extract ID...)\n",
      "\n",
      "Usage:\n",
      "  seqkit seq [flags]\n",
      "\n",
      "Flags:\n",
      "  -k, --color                     colorize sequences - to be piped into \"less -R\"\n",
      "  -p, --complement                complement sequence, flag '-v' is recommended to switch on\n",
      "      --dna2rna                   DNA to RNA\n",
      "  -G, --gap-letters string        gap letters (default \"- \\t.\")\n",
      "  -h, --help                      help for seq\n",
      "  -l, --lower-case                print sequences in lower case\n",
      "  -M, --max-len int               only print sequences shorter than the maximum length (-1 for no limit) (default -1)\n",
      "  -R, --max-qual float            only print sequences with average quality less than this limit (-1 for no limit) (default -1)\n",
      "  -m, --min-len int               only print sequences longer than the minimum length (-1 for no limit) (default -1)\n",
      "  -Q, --min-qual float            only print sequences with average quality qreater or equal than this limit (-1 for no limit) (default -1)\n",
      "  -n, --name                      only print names\n",
      "  -i, --only-id                   print ID instead of full head\n",
      "  -q, --qual                      only print qualities\n",
      "  -b, --qual-ascii-base int       ASCII BASE, 33 for Phred+33 (default 33)\n",
      "  -g, --remove-gaps               remove gaps\n",
      "  -r, --reverse                   reverse sequence\n",
      "      --rna2dna                   RNA to DNA\n",
      "  -s, --seq                       only print sequences\n",
      "  -u, --upper-case                print sequences in upper case\n",
      "  -v, --validate-seq              validate bases according to the alphabet\n",
      "  -V, --validate-seq-length int   length of sequence to validate (0 for whole seq) (default 10000)\n",
      "\n",
      "Global Flags:\n",
      "      --alphabet-guess-seq-length int   length of sequence prefix of the first FASTA record based on which seqkit guesses the sequence type (0 for whole seq) (default 10000)\n",
      "      --id-ncbi                         FASTA head is NCBI-style, e.g. >gi|110645304|ref|NC_002516.2| Pseud...\n",
      "      --id-regexp string                regular expression for parsing ID (default \"^(\\\\S+)\\\\s?\")\n",
      "      --infile-list string              file of input files list (one file per line), if given, they are appended to files from cli arguments\n",
      "  -w, --line-width int                  line width when outputing FASTA format (0 for no wrap) (default 60)\n",
      "  -o, --out-file string                 out file (\"-\" for stdout, suffix .gz for gzipped out) (default \"-\")\n",
      "      --quiet                           be quiet and do not show extra information\n",
      "  -t, --seq-type string                 sequence type (dna|rna|protein|unlimit|auto) (for auto, it automatically detect by the first sequence) (default \"auto\")\n",
      "  -j, --threads int                     number of CPUs. (default value: 1 for single-CPU PC, 2 for others. can also set with environment variable SEQKIT_THREADS) (default 2)\n"
     ]
    }
   ],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
