{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "65493bb4-5213-4d11-9aa3-01518f6b97a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Folder already exists\n",
      "Folder already exists\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'3.1.0'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from pandas import DataFrame\n",
    "import os\n",
    "import os.path as op\n",
    "import sys\n",
    "import csv\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "plt.show()\n",
    "import glob\n",
    "import seaborn as sns\n",
    "import shutil\n",
    "import networkx as nx\n",
    "\n",
    "pd.set_option(\"display.max_rows\", 6)\n",
    "\n",
    "sys.path.insert(0, '/home/jmunson-mcgee/')\n",
    "from JMM_functions import *\n",
    "\n",
    "outdir='/mnt/scgc/simon/microg2p/analyses/GORG_recruitment/CoverM_comparsion'\n",
    "safe_make_dir(outdir)\n",
    "rundir='/mnt/scgc/simon/microg2p/analyses/GORG_recruitment/CoverM_comparsion'\n",
    "os.chdir(rundir)\n",
    "\n",
    "images='/mnt/scgc/simon/microg2p/analyses/GORG_recruitment/CoverM_comparsion'\n",
    "safe_make_dir(images)\n",
    "    \n",
    "matplotlib.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "abc99cef-69ba-47fc-ad9c-8953f0a0b9b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "coverm=glob.glob('/mnt/scgc/simon/microg2p/analyses/GORG_recruitment/CoverM_comparsion/*GoM_SAGs.tsv')\n",
    "\n",
    "SAGdata='/mnt/scgc/simon/microg2p/analyses/20210325_GoM_recluster/Summary_files/FINAL_GoM_Metadata.csv'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "76e72b2b-6640-4e4f-8e58-a1fb7c8001a0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a8f76d21-208d-402b-8dcc-228e32f6dc15",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTDB_classification</th>\n",
       "      <th>relative_genome_size</th>\n",
       "      <th>checkM_estimated_completeness</th>\n",
       "      <th>estimated_total_genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAA164-E04</td>\n",
       "      <td>2.025597</td>\n",
       "      <td>23.349143</td>\n",
       "      <td>5.458512e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAA536-G10</td>\n",
       "      <td>0.936774</td>\n",
       "      <td>39.525536</td>\n",
       "      <td>2.524389e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AG-337-I02</td>\n",
       "      <td>0.741080</td>\n",
       "      <td>46.064667</td>\n",
       "      <td>1.997037e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>0.870133</td>\n",
       "      <td>38.313333</td>\n",
       "      <td>2.344806e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...</td>\n",
       "      <td>1.280005</td>\n",
       "      <td>28.592500</td>\n",
       "      <td>3.449315e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>526</th>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>1.585405</td>\n",
       "      <td>45.960000</td>\n",
       "      <td>4.272298e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>210 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   GTDB_classification  relative_genome_size  \\\n",
       "1                                           AAA164-E04              2.025597   \n",
       "2                                           AAA536-G10              0.936774   \n",
       "3                                           AG-337-I02              0.741080   \n",
       "..                                                 ...                   ...   \n",
       "519  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...              0.870133   \n",
       "521  d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...              1.280005   \n",
       "526  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...              1.585405   \n",
       "\n",
       "     checkM_estimated_completeness  estimated_total_genome_size  \n",
       "1                        23.349143                 5.458512e+06  \n",
       "2                        39.525536                 2.524389e+06  \n",
       "3                        46.064667                 1.997037e+06  \n",
       "..                             ...                          ...  \n",
       "519                      38.313333                 2.344806e+06  \n",
       "521                      28.592500                 3.449315e+06  \n",
       "526                      45.960000                 4.272298e+06  \n",
       "\n",
       "[210 rows x 4 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols=['name', 'checkM_estimated_completeness', 'final_assembly_length', 'GTDB_classification']\n",
    "SAG_data=pd.read_csv(SAGdata, usecols=cols)\n",
    "\n",
    "SAG_data=SAG_data[SAG_data['checkM_estimated_completeness']!= 0.0]\n",
    "SAG_data=SAG_data[SAG_data['GTDB_classification']!= 'Unclassified']\n",
    "SAG_class=SAG_data[['name', 'GTDB_classification']].copy()\n",
    "\n",
    "genome_size=SAG_data.groupby('GTDB_classification', as_index=False)['final_assembly_length', 'checkM_estimated_completeness'].mean()\n",
    "\n",
    "count=SAG_data.groupby('GTDB_classification', as_index=False)['name'].count()\n",
    "count=count.rename(columns={'name':'count'})\n",
    "genome_size=genome_size.merge(count, on='GTDB_classification', how='left')\n",
    "\n",
    "\n",
    "genome_size['estimated_total_genome_size']=genome_size['final_assembly_length']*(100/genome_size['checkM_estimated_completeness'])\n",
    "\n",
    "# limit the genera that I make this calculation to only be the ones with at least 3 cells\n",
    "genome_size=genome_size[genome_size['count']>=3]\n",
    "genome_size['relative_genome_size']=genome_size['estimated_total_genome_size']/genome_size['estimated_total_genome_size'].mean()\n",
    "genome_size.to_csv(os.path.join(outdir, 'genome_size_calculations.csv'))\n",
    "\n",
    "avg_completeness=genome_size['checkM_estimated_completeness'].mean()\n",
    "avg_genome_size=(genome_size['estimated_total_genome_size'].mean())\n",
    "\n",
    "rel_genome_size=genome_size[['GTDB_classification', 'relative_genome_size', 'checkM_estimated_completeness', \n",
    "                             'estimated_total_genome_size']].copy()\n",
    "rel_genome_size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4e1b1f06-2708-4d4c-a16e-9ead61a6ba7e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>GTDB_classification</th>\n",
       "      <th>ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>unmapped</td>\n",
       "      <td>Other</td>\n",
       "      <td>46.937733</td>\n",
       "      <td>43.552130</td>\n",
       "      <td>52.473760</td>\n",
       "      <td>51.032830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AH-269-F19</td>\n",
       "      <td>SCGC-AAA160-P02</td>\n",
       "      <td>0.008955</td>\n",
       "      <td>0.000693</td>\n",
       "      <td>0.008857</td>\n",
       "      <td>0.066070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AH-135-A01</td>\n",
       "      <td>Pelagibacter</td>\n",
       "      <td>0.023162</td>\n",
       "      <td>0.006670</td>\n",
       "      <td>0.005012</td>\n",
       "      <td>0.004562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7516</th>\n",
       "      <td>AM-384-P22</td>\n",
       "      <td>UBA10364</td>\n",
       "      <td>0.010743</td>\n",
       "      <td>0.004844</td>\n",
       "      <td>0.018268</td>\n",
       "      <td>0.042606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7517</th>\n",
       "      <td>AM-384-P23</td>\n",
       "      <td>UBA8337</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000391</td>\n",
       "      <td>0.000430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7518</th>\n",
       "      <td>AM-382-D14</td>\n",
       "      <td>JAACDM01</td>\n",
       "      <td>0.001061</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.005519</td>\n",
       "      <td>0.002545</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7519 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            name GTDB_classification  \\\n",
       "0       unmapped               Other   \n",
       "1     AH-269-F19     SCGC-AAA160-P02   \n",
       "2     AH-135-A01        Pelagibacter   \n",
       "...          ...                 ...   \n",
       "7516  AM-384-P22            UBA10364   \n",
       "7517  AM-384-P23             UBA8337   \n",
       "7518  AM-382-D14            JAACDM01   \n",
       "\n",
       "      ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             46.937733                     \n",
       "1                                              0.008955                     \n",
       "2                                              0.023162                     \n",
       "...                                                 ...                     \n",
       "7516                                           0.010743                     \n",
       "7517                                           0.000006                     \n",
       "7518                                           0.001061                     \n",
       "\n",
       "      ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             43.552130                     \n",
       "1                                              0.000693                     \n",
       "2                                              0.006670                     \n",
       "...                                                 ...                     \n",
       "7516                                           0.004844                     \n",
       "7517                                           0.000000                     \n",
       "7518                                           0.000019                     \n",
       "\n",
       "      ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             52.473760                     \n",
       "1                                              0.008857                     \n",
       "2                                              0.005012                     \n",
       "...                                                 ...                     \n",
       "7516                                           0.018268                     \n",
       "7517                                           0.000391                     \n",
       "7518                                           0.005519                     \n",
       "\n",
       "      ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \n",
       "0                                             51.032830                    \n",
       "1                                              0.066070                    \n",
       "2                                              0.004562                    \n",
       "...                                                 ...                    \n",
       "7516                                           0.042606                    \n",
       "7517                                           0.000430                    \n",
       "7518                                           0.002545                    \n",
       "\n",
       "[7519 rows x 6 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coverm_df=pd.DataFrame()\n",
    "\n",
    "\n",
    "counter=1\n",
    "for file in coverm:\n",
    "    if counter==1:\n",
    "        coverm_df=pd.read_csv(file, sep=\"\\t\")\n",
    "        counter=2\n",
    "    else:\n",
    "        df=pd.read_csv(file, sep=\"\\t\")\n",
    "        coverm_df=coverm_df.merge(df, on='Genome', how='outer')\n",
    "\n",
    "# slice genomes to only incude the SAG identifier    \n",
    "coverm_df['name']=coverm_df['Genome'].str[:10]    \n",
    "coverm_df\n",
    "# merge with the GTDB genus     \n",
    "coverm_df=coverm_df.merge(SAG_class, on='name', how='left')\n",
    "  \n",
    "\n",
    "\n",
    "# move the column to head of list using index, pop and insert\n",
    "cols = list(coverm_df)\n",
    "cols.insert(0, cols.pop(cols.index('GTDB_classification')))\n",
    "cols.insert(0, cols.pop(cols.index('name')))\n",
    "coverm_df = coverm_df.loc[:, cols]\n",
    "    \n",
    "coverm_df=coverm_df.drop(columns='Genome')\n",
    "coverm_df['GTDB_classification']=coverm_df['GTDB_classification'].fillna('Other')\n",
    "coverm_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b311ea78-fa9e-4ae4-9546-485c20919e4e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)',\n",
       " 'ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)',\n",
       " 'ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)',\n",
       " 'ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#generate a list of just the columns representing metagenomes\n",
    "cols = list(coverm_df)\n",
    "cols=cols[2:]\n",
    "cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e7b89989-fdb7-4fcd-95bb-46b5bac3118a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTDB_classification</th>\n",
       "      <th>ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>relative_genome_size</th>\n",
       "      <th>checkM_estimated_completeness</th>\n",
       "      <th>estimated_total_genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAA164-E04</td>\n",
       "      <td>0.001630</td>\n",
       "      <td>0.001096</td>\n",
       "      <td>0.045917</td>\n",
       "      <td>0.013966</td>\n",
       "      <td>2.025597</td>\n",
       "      <td>23.349143</td>\n",
       "      <td>5.458512e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAA536-G10</td>\n",
       "      <td>0.176623</td>\n",
       "      <td>0.243574</td>\n",
       "      <td>0.434991</td>\n",
       "      <td>0.397502</td>\n",
       "      <td>0.936774</td>\n",
       "      <td>39.525536</td>\n",
       "      <td>2.524389e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AG-337-I02</td>\n",
       "      <td>0.000771</td>\n",
       "      <td>0.002253</td>\n",
       "      <td>0.158285</td>\n",
       "      <td>0.022984</td>\n",
       "      <td>0.741080</td>\n",
       "      <td>46.064667</td>\n",
       "      <td>1.997037e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.008736</td>\n",
       "      <td>0.000106</td>\n",
       "      <td>0.870133</td>\n",
       "      <td>38.313333</td>\n",
       "      <td>2.344806e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...</td>\n",
       "      <td>0.000125</td>\n",
       "      <td>0.002979</td>\n",
       "      <td>0.006860</td>\n",
       "      <td>0.000376</td>\n",
       "      <td>1.280005</td>\n",
       "      <td>28.592500</td>\n",
       "      <td>3.449315e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.002043</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>1.585405</td>\n",
       "      <td>45.960000</td>\n",
       "      <td>4.272298e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>211 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   GTDB_classification  \\\n",
       "0                                           AAA164-E04   \n",
       "1                                           AAA536-G10   \n",
       "2                                           AG-337-I02   \n",
       "..                                                 ...   \n",
       "208  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "209  d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...   \n",
       "210  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "     ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             0.001630                     \n",
       "1                                             0.176623                     \n",
       "2                                             0.000771                     \n",
       "..                                                 ...                     \n",
       "208                                           0.000000                     \n",
       "209                                           0.000125                     \n",
       "210                                           0.000034                     \n",
       "\n",
       "     ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             0.001096                     \n",
       "1                                             0.243574                     \n",
       "2                                             0.002253                     \n",
       "..                                                 ...                     \n",
       "208                                           0.000011                     \n",
       "209                                           0.002979                     \n",
       "210                                           0.000012                     \n",
       "\n",
       "     ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             0.045917                     \n",
       "1                                             0.434991                     \n",
       "2                                             0.158285                     \n",
       "..                                                 ...                     \n",
       "208                                           0.008736                     \n",
       "209                                           0.006860                     \n",
       "210                                           0.002043                     \n",
       "\n",
       "     ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             0.013966                     \n",
       "1                                             0.397502                     \n",
       "2                                             0.022984                     \n",
       "..                                                 ...                     \n",
       "208                                           0.000106                     \n",
       "209                                           0.000376                     \n",
       "210                                           0.000439                     \n",
       "\n",
       "     relative_genome_size  checkM_estimated_completeness  \\\n",
       "0                2.025597                      23.349143   \n",
       "1                0.936774                      39.525536   \n",
       "2                0.741080                      46.064667   \n",
       "..                    ...                            ...   \n",
       "208              0.870133                      38.313333   \n",
       "209              1.280005                      28.592500   \n",
       "210              1.585405                      45.960000   \n",
       "\n",
       "     estimated_total_genome_size  \n",
       "0                   5.458512e+06  \n",
       "1                   2.524389e+06  \n",
       "2                   1.997037e+06  \n",
       "..                           ...  \n",
       "208                 2.344806e+06  \n",
       "209                 3.449315e+06  \n",
       "210                 4.272298e+06  \n",
       "\n",
       "[211 rows x 8 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abundance=coverm_df.groupby(['GTDB_classification'], as_index=False)[cols].sum()\n",
    "abundance.to_csv(os.path.join(outdir, 'coverm_all_genera_relative_abundance_90perc_ID.csv'))\n",
    "\n",
    "abundance=abundance.merge(rel_genome_size, on='GTDB_classification', how='left')\n",
    "\n",
    "# replace the genus with \"Other\" for all lineages where I did not calculate an estimated genomes size because there are <3 cells\n",
    "abundance['GTDB_classification']=np.where(abundance['relative_genome_size'].isnull(), 'Other', abundance['GTDB_classification'])\n",
    "\n",
    "# redo the reative abundance\n",
    "abundance=abundance.groupby(['GTDB_classification'], as_index=False)[cols].sum()\n",
    "\n",
    "abundance=abundance.merge(rel_genome_size, on='GTDB_classification', how='left')\n",
    "abundance['relative_genome_size']=abundance['relative_genome_size'].fillna(1)\n",
    "\n",
    "SAR11_rel=abundance[abundance['GTDB_classification'].str.contains('Pelagibacter')]\n",
    "\n",
    "abundance.to_csv(os.path.join(outdir, 'coverm_genus_relative_abundance_90perc_ID.csv'))\n",
    "abundance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f69de7a7-5638-43f2-b74c-d4f130f7fea7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTDB_classification</th>\n",
       "      <th>ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>relative_genome_size</th>\n",
       "      <th>checkM_estimated_completeness</th>\n",
       "      <th>estimated_total_genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAA164-E04</td>\n",
       "      <td>0.000734</td>\n",
       "      <td>0.000502</td>\n",
       "      <td>0.019880</td>\n",
       "      <td>0.006507</td>\n",
       "      <td>2.025597</td>\n",
       "      <td>23.349143</td>\n",
       "      <td>5.458512e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAA536-G10</td>\n",
       "      <td>0.171999</td>\n",
       "      <td>0.241164</td>\n",
       "      <td>0.407226</td>\n",
       "      <td>0.400478</td>\n",
       "      <td>0.936774</td>\n",
       "      <td>39.525536</td>\n",
       "      <td>2.524389e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AG-337-I02</td>\n",
       "      <td>0.000950</td>\n",
       "      <td>0.002820</td>\n",
       "      <td>0.187312</td>\n",
       "      <td>0.029271</td>\n",
       "      <td>0.741080</td>\n",
       "      <td>46.064667</td>\n",
       "      <td>1.997037e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.008805</td>\n",
       "      <td>0.000115</td>\n",
       "      <td>0.870133</td>\n",
       "      <td>38.313333</td>\n",
       "      <td>2.344806e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...</td>\n",
       "      <td>0.000089</td>\n",
       "      <td>0.002159</td>\n",
       "      <td>0.004700</td>\n",
       "      <td>0.000277</td>\n",
       "      <td>1.280005</td>\n",
       "      <td>28.592500</td>\n",
       "      <td>3.449315e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.001130</td>\n",
       "      <td>0.000261</td>\n",
       "      <td>1.585405</td>\n",
       "      <td>45.960000</td>\n",
       "      <td>4.272298e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>211 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   GTDB_classification  \\\n",
       "0                                           AAA164-E04   \n",
       "1                                           AAA536-G10   \n",
       "2                                           AG-337-I02   \n",
       "..                                                 ...   \n",
       "208  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "209  d__Bacteria;p__SAR324;c__SAR324;o__SAR324;f__N...   \n",
       "210  d__Bacteria;p__Verrucomicrobiota;c__Verrucomic...   \n",
       "\n",
       "     ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             0.000734                     \n",
       "1                                             0.171999                     \n",
       "2                                             0.000950                     \n",
       "..                                                 ...                     \n",
       "208                                           0.000000                     \n",
       "209                                           0.000089                     \n",
       "210                                           0.000020                     \n",
       "\n",
       "     ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             0.000502                     \n",
       "1                                             0.241164                     \n",
       "2                                             0.002820                     \n",
       "..                                                 ...                     \n",
       "208                                           0.000011                     \n",
       "209                                           0.002159                     \n",
       "210                                           0.000007                     \n",
       "\n",
       "     ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             0.019880                     \n",
       "1                                             0.407226                     \n",
       "2                                             0.187312                     \n",
       "..                                                 ...                     \n",
       "208                                           0.008805                     \n",
       "209                                           0.004700                     \n",
       "210                                           0.001130                     \n",
       "\n",
       "     ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "0                                             0.006507                     \n",
       "1                                             0.400478                     \n",
       "2                                             0.029271                     \n",
       "..                                                 ...                     \n",
       "208                                           0.000115                     \n",
       "209                                           0.000277                     \n",
       "210                                           0.000261                     \n",
       "\n",
       "     relative_genome_size  checkM_estimated_completeness  \\\n",
       "0                2.025597                      23.349143   \n",
       "1                0.936774                      39.525536   \n",
       "2                0.741080                      46.064667   \n",
       "..                    ...                            ...   \n",
       "208              0.870133                      38.313333   \n",
       "209              1.280005                      28.592500   \n",
       "210              1.585405                      45.960000   \n",
       "\n",
       "     estimated_total_genome_size  \n",
       "0                   5.458512e+06  \n",
       "1                   2.524389e+06  \n",
       "2                   1.997037e+06  \n",
       "..                           ...  \n",
       "208                 2.344806e+06  \n",
       "209                 3.449315e+06  \n",
       "210                 4.272298e+06  \n",
       "\n",
       "[211 rows x 8 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# Now I need to divide by the relative genome size\n",
    "abundance[cols]=abundance[cols].div(abundance.relative_genome_size, axis=0)\n",
    "\n",
    "# and finally divide each value by the sum of the column it is in and multiple by 100\n",
    "# sums=abundance[cols].sum()\n",
    "abundance[cols]=abundance[cols]/abundance[cols].sum()*100\n",
    "abundance.to_csv(os.path.join(outdir, 'coverm_genus_relative_abundance_genome_size_normalized_90perc_ID.csv'))\n",
    "\n",
    "SAR11_norm=abundance[abundance['GTDB_classification'].str.contains('Pelagibacter')]\n",
    "\n",
    "abundance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "61dc8c9d-7ac7-4d58-868f-8b9074a70ea7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTDB_classification</th>\n",
       "      <th>ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)</th>\n",
       "      <th>relative_genome_size</th>\n",
       "      <th>checkM_estimated_completeness</th>\n",
       "      <th>estimated_total_genome_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Pelagibacter</td>\n",
       "      <td>10.144697</td>\n",
       "      <td>7.460345</td>\n",
       "      <td>9.505372</td>\n",
       "      <td>4.012867</td>\n",
       "      <td>0.508892</td>\n",
       "      <td>50.612276</td>\n",
       "      <td>1.371346e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Pelagibacter_A</td>\n",
       "      <td>0.120178</td>\n",
       "      <td>0.394918</td>\n",
       "      <td>1.005482</td>\n",
       "      <td>0.589150</td>\n",
       "      <td>0.447745</td>\n",
       "      <td>49.246981</td>\n",
       "      <td>1.206570e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Pelagibacteraceae</td>\n",
       "      <td>0.000413</td>\n",
       "      <td>0.002901</td>\n",
       "      <td>0.006262</td>\n",
       "      <td>0.062052</td>\n",
       "      <td>0.598134</td>\n",
       "      <td>32.516667</td>\n",
       "      <td>1.611832e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.896587</td>\n",
       "      <td>20.753333</td>\n",
       "      <td>2.416094e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>189</th>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>0.000219</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.530596</td>\n",
       "      <td>38.756667</td>\n",
       "      <td>1.429832e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Alphaproteoba...</td>\n",
       "      <td>0.002839</td>\n",
       "      <td>0.011117</td>\n",
       "      <td>0.040482</td>\n",
       "      <td>0.018078</td>\n",
       "      <td>0.392090</td>\n",
       "      <td>57.198889</td>\n",
       "      <td>1.056592e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   GTDB_classification  \\\n",
       "69                                        Pelagibacter   \n",
       "70                                      Pelagibacter_A   \n",
       "71                                   Pelagibacteraceae   \n",
       "..                                                 ...   \n",
       "188  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...   \n",
       "189  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...   \n",
       "190  d__Bacteria;p__Proteobacteria;c__Alphaproteoba...   \n",
       "\n",
       "     ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "69                                           10.144697                     \n",
       "70                                            0.120178                     \n",
       "71                                            0.000413                     \n",
       "..                                                 ...                     \n",
       "188                                           0.000014                     \n",
       "189                                           0.000037                     \n",
       "190                                           0.002839                     \n",
       "\n",
       "     ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "69                                            7.460345                     \n",
       "70                                            0.394918                     \n",
       "71                                            0.002901                     \n",
       "..                                                 ...                     \n",
       "188                                           0.000007                     \n",
       "189                                           0.000068                     \n",
       "190                                           0.011117                     \n",
       "\n",
       "     ALL_20181030_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "69                                            9.505372                     \n",
       "70                                            1.005482                     \n",
       "71                                            0.006262                     \n",
       "..                                                 ...                     \n",
       "188                                           0.000030                     \n",
       "189                                           0.000219                     \n",
       "190                                           0.040482                     \n",
       "\n",
       "     ALL_20171130_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)  \\\n",
       "69                                            4.012867                     \n",
       "70                                            0.589150                     \n",
       "71                                            0.062052                     \n",
       "..                                                 ...                     \n",
       "188                                           0.000006                     \n",
       "189                                           0.000029                     \n",
       "190                                           0.018078                     \n",
       "\n",
       "     relative_genome_size  checkM_estimated_completeness  \\\n",
       "69               0.508892                      50.612276   \n",
       "70               0.447745                      49.246981   \n",
       "71               0.598134                      32.516667   \n",
       "..                    ...                            ...   \n",
       "188              0.896587                      20.753333   \n",
       "189              0.530596                      38.756667   \n",
       "190              0.392090                      57.198889   \n",
       "\n",
       "     estimated_total_genome_size  \n",
       "69                  1.371346e+06  \n",
       "70                  1.206570e+06  \n",
       "71                  1.611832e+06  \n",
       "..                           ...  \n",
       "188                 2.416094e+06  \n",
       "189                 1.429832e+06  \n",
       "190                 1.056592e+06  \n",
       "\n",
       "[12 rows x 8 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAR11_norm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2582fb2d-1b21-4acc-81b6-2236a5ea48bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GTDB_classification                                                    PelagibacterPelagibacter_APelagibacteraceaed__...\n",
       "ALL_20190709_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)                                              10.4656\n",
       "ALL_20190402_contf_pe_bbmerge_reads.fastq.gz Relative Abundance (%)                                              8.24109\n",
       "                                                                                             ...                        \n",
       "relative_genome_size                                                                                             6.52492\n",
       "checkM_estimated_completeness                                                                                    533.173\n",
       "estimated_total_genome_size                                                                                  1.75831e+07\n",
       "Length: 8, dtype: object"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SAR11_norm.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60cbec9d-cb6c-41d2-9ba1-44885d5c22f0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Jacob",
   "language": "python",
   "name": "jacob"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
